src/coreclr-debug.ts

```diff
@@ -0,0 +1,174 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import * as child_process from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+
+let _coreClrDebugDir: string;
+let _debugAdapterDir: string;
+let _channel: vscode.OutputChannel;
+let _installLog: NodeJS.WritableStream;
+const _completionFileName: string = 'install.complete';
+
+export function installCoreClrDebug(context: vscode.ExtensionContext) {
+    _coreClrDebugDir = path.join(context.extensionPath, 'coreclr-debug');
+    _debugAdapterDir = path.join(_coreClrDebugDir, 'debugAdapters');
+    
+    if (existsSync(path.join(_debugAdapterDir, _completionFileName))) {
+        console.log('.NET Core Debugger tools already installed');
+        return;
+    }
+    
+    _channel = vscode.window.createOutputChannel('coreclr-debug');
+    
+    // Create our log file and override _channel.append to also outpu to the log
+    _installLog = fs.createWriteStream(path.join(_coreClrDebugDir, 'install.log'));
+    (function() {
+        var proxied = _channel.append;
+        _channel.append = function(val: string) {
+            _installLog.write(val);
+            proxied.apply(this, arguments);
+        };
+    })();
+
+    _channel.appendLine("Downloading and configuring the .NET Core Debugger...");
+    _channel.show(vscode.ViewColumn.Three);
+    
+    spawnChildProcess('dotnet', ['--verbose', 'restore'], _channel, _coreClrDebugDir)
+    .then(function() {
+        return spawnChildProcess('dotnet', ['--verbose', 'publish', '-o', _debugAdapterDir], _channel, _coreClrDebugDir);
+    }).then(function() {
+        var promises: Promise<void>[] = [];
+
+        promises.push(renameDummyEntrypoint());
+        promises.push(removeLibCoreClrTraceProvider());
+
+        return Promise.all(promises);
+    }).then(function() {
```

Comment:
Should we either check the version of 'dotnet' or check that it created the content file items to handle the case that the user has an older version of .NET CLI installed?